source/utils/HexagonMeshTools.cc

```diff
@@ -0,0 +1,99 @@
+// ----------------------------------------------------------------------------
+// nexus | HexagonMeshTools.cc
+//
+// This class provides the functions needed to create a hexagonal mesh
+//
+// The NEXT Collaboration
+// ----------------------------------------------------------------------------
+
+#include "HexagonMeshTools.h"
+#include <G4SubtractionSolid.hh>
+#include <G4Polyhedra.hh>
+#include <G4PVPlacement.hh>
+
+
+namespace nexus {
+
+  using namespace CLHEP;
+
+
+
+  G4ExtrudedSolid* HexagonMeshTools::CreateHexagon(G4double Thickness, G4double circumradius){
+    
+    // Define a hexagonal prism
+    const G4int nsect = 6;
+    
+    // Create the nodes of the hexagon
+    std::vector<G4TwoVector> polygon(nsect);
+    G4double ang = 2*pi/nsect;
+    for (G4int i = 0; i < nsect; ++i) {
+        G4double phi = i*ang;
+        G4double cosphi = std::cos(phi);
+        G4double sinphi = std::sin(phi);
+        polygon[i].set(circumradius*cosphi, circumradius*sinphi);
+    }
+    
+    // Define the hexagon
+    G4TwoVector offsetA(0,0), offsetB(0,0);
+    G4double scaleA = 1, scaleB = 1;
+    return new G4ExtrudedSolid("Extruded", polygon, Thickness, offsetA, scaleA, offsetB, scaleB);
+
+  }
+
+  void HexagonMeshTools::PlaceHexagons(G4int nHole, G4double InRadius, G4double Thickness, G4LogicalVolume* DiskLogical, G4LogicalVolume* HexLogical, G4double Mesh_D){
+    
+    // Logical Volume of SS Disk
+
+    // Use cube coordinate system defined in:
+    // https://www.redblobgames.com/grids/hexagons/
+
+    // Dist from hexagon centre to center of hexagon vertex (inc. wire thickness)
+    G4double HexSize = (InRadius + Thickness)/std::sqrt(3.0)*mm;
```

Comment:
Why are you multiplying by `mm`? In principle both `InRadius` and `Thickness` have already their units included.

Reply:
Again, I got lucky here, and `*mm` multiplies by 1. I have removed it.